ta4j-core/src/main/java/org/ta4j/core/Decimal.java

```diff
@@ -72,24 +74,28 @@ private Decimal(String val) {
         delegate = new BigDecimal(val, MATH_CONTEXT);
     }
 
-    /**
-     * Constructor.
-     * @param val the double value
-     */
-    private Decimal(double val) {
-        delegate = BigDecimal.valueOf(val);
+    private Decimal(short val) {
+        delegate = new BigDecimal(val, MATH_CONTEXT);
     }
 
     private Decimal(int val) {
-        delegate = BigDecimal.valueOf(val);
+        delegate = new BigDecimal(val, MATH_CONTEXT);
     }
 
     private Decimal(long val) {
-        delegate = BigDecimal.valueOf(val);
+        delegate = new BigDecimal(val, MATH_CONTEXT);
+    }
+
+    private Decimal(float val) {
+        delegate = new BigDecimal(val, MATH_CONTEXT);
+    }
+
+    private Decimal(double val) {
+        delegate = new BigDecimal(val, MATH_CONTEXT);
     }
```

Comment:
Please use static function `valueOf(val)` to generate new Decimal instances. See  #84 and #86

Reply:
This is a constructor

Reply:
Oh, nevermind - i see what you mean.